semeio/workflows/localisation/local_script_lib.py

```diff
@@ -220,33 +227,85 @@ def write_qc_parameter(
         if param_for_field is None or field_scale is None:
             return
 
+        # Write scaling parameter  once per corr_name
+        if corr_name == cls.corr_name:
+            return
+
+        cls.corr_name = corr_name
+
+        # Need a parameter name <= 8 character long for GRDECL format
+        scaling_kw_name = "S_" + str(cls.scaling_param_number)
+        filename = cls.corr_name + "_" + node_name + "_" + scaling_kw_name + ".roff"
+
         scaling_values = np.reshape(
-            param_for_field, (grid.getNX(), grid.getNY(), grid.getNZ()), "F"
+            param_for_field, (grid.getNX(), grid.getNY(), grid.getNZ()), "C"
+        )
+        save_field(scaling_values, scaling_kw_name, filename, FieldFileFormat.ROFF)
+
+        print(
+            "Write calculated scaling factor  with name: "
+            f"{scaling_kw_name} to file: {filename}"
+        )
+        debug_print(
+            f"Write calculated scaling factor with name: "
+            f"{scaling_kw_name} to file: {filename}",
+            LogLevel.LEVEL3,
+            log_level,
         )
 
+        # Increase parameter number to define unique parameter name
+        cls.scaling_param_number = cls.scaling_param_number + 1
+
+    @classmethod
+    def write_qc_parameter_surface(
+        # pylint: disable=too-many-locals
+        cls,
+        node_name: str,
+        corr_name: str,
+        surface_scale: bool,
+        reference_surface_file: str,
+        param_for_surface: ma.MaskedArray,
+        log_level: LogLevel = LogLevel.OFF,
+    ) -> None:
+        # pylint: disable=too-many-arguments
+
+        if param_for_surface is None or surface_scale is None:
+            return
+
         # Write scaling parameter  once per corr_name
-        if corr_name != cls.corr_name:
-            cls.corr_name = corr_name
-            # Need a parameter name <= 8 character long
-            scaling_kw_name = "S_" + str(cls.scaling_param_number)
-            scaling_kw = grid.create_kw(scaling_values, scaling_kw_name, False)
-            filename = (
-                cls.corr_name + "_" + node_name + "_" + scaling_kw_name + ".GRDECL"
-            )
-            print(
-                "Write calculated scaling factor  with name: "
-                f"{scaling_kw_name} to file: {filename}"
-            )
-            debug_print(
-                f"Write calculated scaling factor with name: "
-                f"{scaling_kw_name} to file: {filename}",
-                LogLevel.LEVEL3,
-                log_level,
-            )
-            with cwrap.open(filename, "w") as file:
-                grid.write_grdecl(scaling_kw, file)
-            # Increase parameter number to define unique parameter name
-            cls.scaling_param_number = cls.scaling_param_number + 1
+        if corr_name == cls.corr_name:
+            return
+
+        cls.corr_name = corr_name
```

Comment:
This is probably not a good idea, as it has some wierd side effects, take for example:
```python
class MyClass:
    foo = 1

    @classmethod
    def bar(cls):
        if cls.foo == 1:
            print(1)
        else:
            print(1)
        cls.foo = 2

test = MyClass()
print(test.foo)
>>> 1

MyClass.bar()

print(test.foo)
>>> 2
```

Notice that this last print is 2, even though `test` has never called `bar`

it even means that if we later do:

```python
test_2 = MyClass()

print(test_2.foo)
>>> 2
```

which will be very hard to track down if we ever need to debug it.

Reply:
It is possible to remove the class ScalingValues and let the function add_ministeps count the number of fields and surface parameters used in a ministep to avoid writing the qc data more than once per ministep per type (field, surface)

Reply:
I will propose an new solution without the ScalingValues class.